src/DataType.cpp

```diff
@@ -74,6 +74,15 @@ DataType string_to_data_type(const std::string& dtype) {
     return type_map[dtype_l];
 }
 
+
+bool data_type_is_numeric(DataType dtype) {
+    return !(dtype == DataType::Bool ||
+             dtype == DataType::Char ||
+             dtype == DataType::String ||
+             dtype == DataType::Nothing);
+}
```

Comment:
`::Opaque`, `::DateTime` and `Date` should probably be in that list too, no? Probably saver to whitelist because if we extend the list and forget to add it here we might be in trouble.